GenericArrayTests/GenericArrayTests.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace GenericArrayTests
+{
+    class GenericArrayTests
+    {
+        [TestClass]
+        public class ArrayTests
+        {
+            [DataTestMethod]
+            [DataRow(1)]
+            [DataRow(2)]
+            [DataRow(3)]
+            public void Add_Success(int item)
```

Comment:
According to [Microsoft's unit testing best practices](https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices), you might want the name of your test(s) to consist of three parts:

The name of the method being tested.
The scenario under which it's being tested.
The expected behavior when the scenario is invoked.
```suggestion
            public void Add_OneItem_Success(int item)
```

Reply:
Best practices also say to use one assert per test. I will split them up next time.

More typing, though.... :D